### DISCUSSION TITLE
Routing Error

### DISCUSSION THREAD

**Discussion by PhilipAmos** [sandermvanvliet/RoadCaptain#83] [General]
I have noticed a routing error.  If you are travelling as follows:
“Sequoia to Marina”
Straight to “Marina”
Left to “Italian villas loop 1”.

You are not at the junction where in Zwift you can choose the following:
Left to "Volcano to the Italian villas" (takes you to Volcano)
Right to "Italian villas loop 1" (takes you to the sprint)

The problem is RoadCaptain thinks the choices are Left or Straight and not Left or Right.  This is an excerpt for the JSON file for the incorrect junction showing it tries to route Straight.

      {
        "segmentId": "watopia-bambino-fondo-001-after-after-after-before",
        "turnToNextSegment": "goStraight",
        "nextSegmentId": "watopia-bambino-fondo-001-after-after-after-after-before-before",
        "direction": "atoB"
      },

If you change the ”goStraight” to “right” it works correctly otherwise it takes the left turn towards the Volcano.

**Comment by sandermvanvliet** [comment]
Closing this topic as the routing now handles the turns a lot better than when this topic was started.
If routing errors pop up please create an issue 👍
